source/grid/grid_out.cc

```diff
@@ -2526,6 +2529,107 @@ void GridOut::write_vtu (const Triangulation<dim,spacedim> &tria,
+  data_names.push_back("lvlsubdomain");
+  data_names.push_back("level");
+  data_names.push_back("proc_writing");
+
+  unsigned int n_q_points=GeometryInfo<dim>::vertices_per_cell;
+
+  typename Triangulation<dim, spacedim>::cell_iterator cell, endc;
+  for (cell=tria.begin(), endc=tria.end();
+       cell != endc; ++cell)
+    {
+      if (!include_artificial && cell->level_subdomain_id() ==
+          numbers::artificial_subdomain_id)
+        continue;
+      if (!view_levels && cell->has_children())
+        continue;
+
+      DataOutBase::Patch<dim,spacedim> patch;
+      patch.data.reinit(n_datasets, n_q_points);
+      patch.points_are_available = false;
+
+      for (unsigned int vertex=0; vertex<n_q_points; ++vertex)
+        {
+          patch.vertices[vertex] = cell->vertex(vertex);
+          patch.data(0,vertex) = cell->level();
+          if (!cell->has_children())
+            patch.data(1,vertex) =
+              (double)static_cast<int>(cell->subdomain_id());
+          else
+            patch.data(1,vertex) = -1.0;
+          patch.data(2,vertex) =
+            (double)static_cast<int>(cell->level_subdomain_id());
+          patch.data(3,vertex) =
+            (double)static_cast<int>(Utilities::MPI::this_mpi_process(MPI_COMM_WORLD));
+        }
+
+      for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+        patch.neighbors[f] = numbers::invalid_unsigned_int;
+      patches.push_back(patch);
+    }
+
+  const std::string new_file = (filename + ".proc" +
+                                Utilities::int_to_string (tria.locally_owned_subdomain(), 4) +
+                                ".vtu");
+  std::ofstream out(new_file.c_str());
+  std::vector<std_cxx1x::tuple<unsigned int, unsigned int, std::string> > vector_data_ranges;
+  DataOutBase::VtkFlags flags;
+  DataOutBase::write_vtu (patches,
+                          data_names,
+                          vector_data_ranges,
+                          flags,
+                          out);
+  //creates .pvtu record
+  if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0)
+    {
+      std::vector<std::string> filenames;
+      unsigned int indx = filename.size();
+
+      //.pvtu file must exclude path/to/filename
+      for (std::string::iterator it = filename.end();
+           it != filename.begin(); --it)
+        {
+          if (*it == '/')
+            break;
+          indx -= 1;
+        }
+      for (unsigned int i=0; i<Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD); ++i)
+        filenames.push_back (filename.substr(indx+1,filename.size()) +
+                             ".proc" + Utilities::int_to_string(i, 4) +
+                             ".vtu");
+
+      const std::string pvtu_master_filename = (filename + ".pvtu");
+      std::ofstream pvtu_master (pvtu_master_filename.c_str());
+
+      DataOut<dim, DoFHandler<dim,spacedim> > data_out;
+      data_out.attach_triangulation (tria);
+      Vector<float> subdomain (tria.n_active_cells());
```

Comment:
This is a dummy variable. Can you change the name to something more appropriate (dummy_field or something) and add a comment?